lib/plugins/physics.js

```diff
@@ -196,26 +195,37 @@ function inject (bot) {
     })
   }
 
+  let lookingTask = createDoneTask()
+
+  bot.on('move', () => {
+    if (Math.abs(deltaYaw(bot.entity.yaw, lastSentYaw)) < 0.001 && !lookingTask.done) {
+      lookingTask.finish()
+    }
+  })
+
   bot.look = callbackify(async (yaw, pitch, force) => {
+    if (!lookingTask.done) {
+      lookingTask.finish() // finish the previous one
+    }
+    lookingTask = createTask()
+
     bot.entity.yaw = yaw
     bot.entity.pitch = pitch
     if (force) {
       lastSentYaw = yaw
       return
     }
 
-    do {
-      await once(bot, 'move')
-    } while (Math.abs(deltaYaw(bot.entity.yaw, lastSentYaw)) >= 0.001)
+    await lookingTask.promise
   })
 
-  bot.lookAt = callbackify(async (point, force) => {
+  bot.lookAt = (point, force, cb) => {
```

Comment:
why did you add back the cb ?

Reply:
Because this is just the lookAt function, it is a proxy function for the look function. All the functions still take cb, but this one I made it just return the promise of look() after calculating those values (yaw and pitch). Otherwise it needed an await in there.

Reply:
what's the problem for having an await ?

Reply:
Not really a problem, this just seemed the most pragmatic, it being a proxy function it doesn't really need to be async.

Reply:
the problem with doing it that way is that it will make it harder to remove callbacks the day we want to
I think it's better in mineflayer internals to never call functions with their callbacks
and to callbackify them from the outside
That way removing the callback can be done simply by removing the callbackify, nothing else

Reply:
Ok fine, done